### PR TITLE
docs: adding algokit template documentation

### PR DESCRIPTION
**Pull request by negar-abbasi**

### CODE REVIEW

File: docs/tutorials/algokit-template.md
Comment (aorumbayev) [inline docs/tutorials/algokit-template.md:97]: AlgoKit templates don't include bootstrap by itself instead its extra logic that algokit-cli performs itself as part of the logic executed during algokit init. 

Something like this instead could be a bit more explicit:
```
When instantiating your template via AlgoKit CLI it will optionally prompt the user to automatically run [algokit bootstrap](put link to bootstrap command docs from this folder) after the project is initialized and can perform various setup tasks like installing dependencies or setting up databases.```

File: docs/tutorials/algokit-template.md
Comment (aorumbayev) [inline docs/tutorials/algokit-template.md:99]: My upcoming pr will introduce some extra behaviour here. Something like below should be a better match:

`env`: Copies an `.env*.template` file to `.env*` in the current working directory and prompts for any unspecified values. This is a critical aspect of managing environment variables securely, ensuring that sensitive data doesn't accidentally end up in version control. 

>Please note by default algokit will scan for network prefixed `.env` variables (for example `.env.localnet` and etc). If no such prefixed files are found it will then attempt to load `.env` files by default. The network prefixed env variables are useful for cases when you want to rely on [algokit deploy command](put link to deploy command documentation once its merged from https://github.com/algorandfoundation/algokit-cli/blob/deploy-command/docs/features/deploy.md)

File: docs/tutorials/algokit-template.md
Comment (aorumbayev) [inline docs/tutorials/algokit-template.md:120]: To be clear these default behaviours aren't included into the templates, these are coming from algokit-cli itself and can be disabled by the user creating a template during algokit init via custom flags (see algokit init --help)

So perhaps slightly more explicit wording here:
```
When creating an AlgoKit template, there are a few default behaviors that you can expect to be provided by algokit-cli itself without introducing any extra code to your templates:
```

File: docs/tutorials/algokit-template.md
Comment (aorumbayev) [inline docs/tutorials/algokit-template.md:122]: In this case since this behavior is handled by algokit cli on top of copier i think its better to rename mentions of copier (like `Copier will commit...) to `AlgoKit CLI will commit` and etc

File: docs/tutorials/algokit-template.md
Comment (aorumbayev) [inline docs/tutorials/algokit-template.md:124]: C̶o̶p̶i̶e̶r̶ -> AlgoKit CLI will automatically open the newly created VSCode window unless user provides specific flags into the init command

File: docs/tutorials/algokit-template.md
Comment (aorumbayev) [inline docs/tutorials/algokit-template.md:126]: Similar to previous comments, bootsrap is handled by algokit cli its not controlled from within the codebase of the template that user creates, please sligtly rephrase to make it clear who is executing the bootstrap

File: docs/tutorials/algokit-template.md
Comment (aorumbayev) [inline docs/tutorials/algokit-template.md:136]: Please also include link to https://github.com/algorandfoundation/algokit-cli/blob/main/docs/algokit.md#guiding-principles

File: docs/tutorials/algokit-template.md
Comment (aorumbayev) [inline docs/tutorials/algokit-template.md:139]: Like the conclusion 🚀

**Review by aorumbayev — CHANGES_REQUESTED** [review]

File: docs/tutorials/algokit-template.md
Comment (aorumbayev) [inline docs/tutorials/algokit-template.md:20]: Let's add another section called quick start in here before any other sections, because despite overview of the templates itself we still don't directly provide a guide on how exactly to create them for users. So it will serve a quick TL;DR section that users runs over before he dives into reading the tutorial because he will need the reference beaker template anyway.

Since we are assuming beaker templates lets do a simple set of numbered items on things that a template builder needs to setup before he reads the remaining part of this tutorial. 

Something like:

```
For users who are keen on getting started with creating AlgoKit templates, you can follow these quick steps:

1. Click on `Use this template`->`Create a new repository` on [algokit-beaker-default-template](https://github.com/algorandfoundation/algokit-beaker-default-template) Github page. This will create a new reference repository with clean git history, allowing you to start modifying and transforming the base beaker template into your own custom template. 
2. Modify the cloned template according to your specific needs. You can refer to the remainder of this tutorial for an understanding of expected behaviors from the AlgoKit side, Copier - the templating framework, and key concepts related to the default files you will encounter in the reference template.
```

@Loedn please note merging this PR assumes we turned the repos to template repos (at least the beaker repository) so this ways folks wouldn't need to Clone-> Clear git history each time but just make a templage repo from base template repo.

**Review by aorumbayev — CHANGES_REQUESTED** [review]
Just final set of comments from my side and we are good to go for this initial version of the tutorial. 

@joe-p feel free to check as well if you have any further suggestions before we merge this. The goal here is to set a starting point for a more detailed doc for template builders (currently we only have a few paragraphs on algokit docs that primarily just forward to beaker repo or copier docs). This doc is set to set more explicit context on templates structure and etc.

**Review by aorumbayev — CHANGES_REQUESTED** [review]
@negar-abbasi aside from minor typo comment just one extra ref I think we would want to include is:

On this doc https://github.com/algorandfoundation/algokit-cli/blob/main/docs/features/init.md#community-templates, please modify the:
`If you want to create a community template, you can use the [beaker template](https://github.com/algorandfoundation/algokit-beaker-default-template) and [Copier documentation](https://copier.readthedocs.io/en/stable/) as a starting point.` to instead point out to this document because that essentially expands a lot on how templates work and how to quick start making your own.  

Once addressed I'll approve the changes.

@Loedn so i think devrel is ok with using the templates feature? If yes lmk if you can approve this once template repos have template toggle enabled and we can merge this

File: docs/features/init.md
Comment (aorumbayev) [inline docs/features/init.md:95]:
```suggestion
If you want to create a community template, you can use [our guidelines on template building](https://github.com/algorandfoundation/algokit-cli/blob/main/docs/tutorials/algokit-template.md#creating-algokit-templates) and [Copier documentation](https://copier.readthedocs.io/en/stable/) as a starting point.
```

**Review by aorumbayev — APPROVED** [review]

**Comment by Loedn** [comment]
all clear on my end

**Comment by aorumbayev** [comment]
@Loedn please keep in mind you'd need to turn full stack, react and beaker repos into template repos on github once 1.3.0 release is out

**Pull request merged**
